lib/src/int64_native.dart

```diff
@@ -0,0 +1,376 @@
+
+  @override
+  Int64 operator ^(Object other) => Int64(_i ^ _promote(other));
+
+  @override
+  Int64 operator ~() => Int64(~_i);
+
+  @override
+  Int64 operator <<(int shiftAmount) => Int64(_i << shiftAmount);
+
+  @override
+  Int64 operator >>(int shiftAmount) => Int64(_i >> shiftAmount);
+
+  @override
+  Int64 shiftRightUnsigned(int shiftAmount) => Int64(_i >>> shiftAmount);
+
+  @override
+  int compareTo(Object other) => _i.compareTo(_promote(other));
+
+  @override
+  bool operator <(Object other) => _i < _promote(other);
+
+  @override
+  bool operator <=(Object other) => _i <= _promote(other);
+
+  @override
+  bool operator >(Object other) => _i > _promote(other);
+
+  @override
+  bool operator >=(Object other) => _i >= _promote(other);
+
+  @override
+  bool get isEven => _i.isEven;
+
+  @override
+  bool get isMaxValue => this == MAX_VALUE;
+
+  @override
+  bool get isMinValue => this == MIN_VALUE;
+
+  @override
+  bool get isNegative => _i < 0;
+
+  @override
+  bool get isOdd => _i.isOdd;
+
+  @override
+  bool get isZero => _i == 0;
+
+  @override
+  int get bitLength => _i.bitLength;
+
+  /// Returns a hash code based on all the bits of this [Int64].
+  @override
+  int get hashCode => _i.hashCode;
+
+  /// Returns [:true:] if this [Int64] has the same numeric value as the given
+  /// object.  The argument may be an [int] or an [IntX].
+  @override
+  bool operator ==(Object other) => _i == _promote(other);
+
+  @override
+  Int64 abs() => Int64(_i.abs());
+
+  @override
+  Int64 clamp(Object lowerLimit, Object upperLimit) =>
+      Int64(_i.clamp(_promote(lowerLimit), _promote(upperLimit)));
+
+  /// Returns the number of leading zeros in this [Int64] as an [int]
+  /// between 0 and 64.
+  @override
+  int numberOfLeadingZeros() {
+    final high = _i >>> 32;
+    final highLeadingZeroes = u.numberOfLeadingZeros(high);
+    if (highLeadingZeroes == 32) {
+      final low = _i & 0xFFFFFFFF;
+      final lowLeadingZeroes = u.numberOfLeadingZeros(low);
+      return 32 + lowLeadingZeroes;
+    }
+    return highLeadingZeroes;
+  }
+
+  /// Returns the number of trailing zeros in this [Int64] as an [int] between
+  /// 0 and 64.
+  @override
+  int numberOfTrailingZeros() {
+    final low = _i & 0xFFFFFFFF;
+    final lowTrailingZeros = u.numberOfTrailingZeros(low);
+    if (lowTrailingZeros == 32) {
+      final high = _i >>> 32;
+      final highTrailingZeros = u.numberOfTrailingZeros(high);
+      return 32 + highTrailingZeros;
+    }
+    return lowTrailingZeros;
+  }
+
+  @override
+  Int64 toSigned(int width) {
+    if (width < 1 || width > 64) {
+      throw RangeError.range(width, 1, 64);
+    }
+    return Int64(_i.toSigned(width));
+  }
+
+  @override
+  Int64 toUnsigned(int width) {
+    if (width < 0 || width > 64) {
+      throw RangeError.range(width, 0, 64);
+    }
+    return Int64(_i.toUnsigned(width));
+  }
+
+  @override
+  List<int> toBytes() {
+    final result = List<int>.filled(8, 0);
```

Comment:
(Really should be returning a `Uint8List` here. Nobody ever wants anything else. If anyone ever uses this for anything. A `writeBytes(List<int> target, int offset)` would be more useful.)

Reply:
I agree this should be `writeBytes`, it makes no sense to serialize an int to a fresh array as you'll always be copying it to another array.